pkg/cmd/experimental/ipfailover/ipfailover.go

```diff
@@ -163,18 +173,52 @@ func Run(f *clientcmd.Factory, options *ipfailover.IPFailoverConfigCmdOptions, c
 		return err
 	}
 
-	if options.Action.ShouldPrint() {
-		mapper, _ := f.Object(false)
-		return cmdutil.VersionedPrintObject(f.PrintObject, cmd, mapper, options.Action.Out)(list)
-	}
-
 	namespace, _, err := f.DefaultNamespace()
 	if err != nil {
 		return err
 	}
+	_, kClient, err := f.Clients()
+	if err != nil {
+		return fmt.Errorf("error getting client: %v", err)
+	}
+	if err := validateServiceAccount(kClient, namespace, options.ServiceAccount); err != nil {
+		return fmt.Errorf("ipfailover could not be created; %v", err)
+	}
+
+	configList := []runtime.Object{
+		&kapi.ServiceAccount{ObjectMeta: kapi.ObjectMeta{Name: options.ServiceAccount}},
+	}
+
+	list.Items = append(configList, list.Items...)
+
+	if options.Action.ShouldPrint() {
+		mapper, _ := f.Object(false)
+		return cmdutil.VersionedPrintObject(f.PrintObject, cmd, mapper, options.Action.Out)(list)
+	}
 
 	if errs := options.Action.WithMessage(fmt.Sprintf("Creating IP failover %s", name), "created").Run(list, namespace); len(errs) > 0 {
 		return cmdutil.ErrExit
 	}
 	return nil
 }
+
+func validateServiceAccount(client *kclient.Client, ns string, serviceAccount string) error {
+
+	sccList, err := client.SecurityContextConstraints().List(kapi.ListOptions{})
+	if err != nil {
+		if !errors.IsUnauthorized(err) {
+			return fmt.Errorf("could not retrive list of security constraints to verify service account %q: %v", serviceAccount, err)
+		}
+	}
+
+	for _, scc := range sccList.Items {
+		if scc.ObjectMeta.Name == "privileged" {
```

Comment:
don't check a specific name, check if it has the required permissions (AllowPrivilegedContainer, host mount, host network, etc). Ideally, this would actually use the new API to just check if the proposed podspec could be created using this service account, and not even do the scc check directly, but I'm not sure if that API is functional yet